File: lib/domain/crop/dtos/crop_ai_create_failure.rb
# frozen_string_literal: true

module Domain
  module Crop
    module Dtos
      class CropAiCreateFailure
        attr_reader :http_status, :message

        def initialize(http_status:, message:)
          @http_status = http_status
          @message = message
        end
      end
    end
  end
end

File: crates/agrr-domain/src/crop/dtos/crop_ai_create_failure.rs
use crate::crop::dtos::HttpStatus;

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct CropAiCreateFailure {
    pub http_status: HttpStatus,
    pub message: String,
}

impl CropAiCreateFailure {
    pub fn new(http_status: HttpStatus, message: impl Into<String>) -> Self {
        Self { http_status, message: message.into() }
    }
}
